From 735209bfe64b6b72132b7038409e0cd86e4aa686 Mon Sep 17 00:00:00 2001
From: JS Deck <jsdeckerido@gmail.com>
Date: Thu, 11 Dec 2025 01:30:05 -0400
Subject: [PATCH 12/14] winepulse: Typo and minor cleanup.

---
 dlls/winepulse.drv/pulse.c | 29 +++++++++++++++--------------
 1 file changed, 15 insertions(+), 14 deletions(-)

diff --git a/dlls/winepulse.drv/pulse.c b/dlls/winepulse.drv/pulse.c
index ecf80bdb724..56dbd262a28 100644
--- a/dlls/winepulse.drv/pulse.c
+++ b/dlls/winepulse.drv/pulse.c
@@ -824,10 +824,10 @@ static GUID get_container_id(const char *sysfs_path)
     create_usb_dev_container_id(init_time, vid, pid, bus_num, dev_num, &tmp_guid);
 
 exit:
-    if (udev)
-        udev_unref(udev);
     if (audio_dev)
         udev_device_unref(audio_dev);
+    if (udev)
+        udev_unref(udev);
 
     TRACE("Returning %s.\n", debugstr_guid(&tmp_guid));
     return tmp_guid;
@@ -975,7 +975,7 @@ static void pulse_stream_probe_cb(pa_stream *stream, void *user)
                 pa_stream_unref(stream);
                 dev->probe_status = -1;
                 pulse_dev_set_ready(dev);
-                TRACE("probe error: %px, %d\n", dev, state);
+                TRACE("probe error: %p, %d\n", dev, state);
             }
         }
     } else if (dev->probe_status == PA_STREAM_READY) {
@@ -1211,26 +1211,27 @@ static void pulse_contextcallback(pa_context *c, void *userdata)
 
         case PA_CONTEXT_READY:
             TRACE("Ready\n");
-            pa_context_set_subscribe_callback(c, pulse_subscribecallback, userdata);
-            o = pa_context_subscribe(
-                c,
-                (pa_subscription_mask_t)(PA_SUBSCRIPTION_MASK_SINK | PA_SUBSCRIPTION_MASK_SOURCE),
-                NULL,
-                userdata);
-            if (o)
-                pa_operation_unref(o);
-
             free_phys_device_lists();
             /* Burnout Paradise Remastered expects device name to have a space. */
             pulse_add_device(eRender, NULL, 0, Speakers, 0, "", "PulseAudio Output");
             pulse_add_device(eCapture, NULL, 0, Microphone, 0, "", "PulseAudio Input");
 
-            o = pa_context_get_sink_info_list(pulse_ctx, &pulse_phys_speakers_cb, NULL);
+            o = pa_context_get_sink_info_list(c, &pulse_phys_speakers_cb, NULL);
             if (o)
                 pa_operation_unref(o);
-            o = pa_context_get_source_info_list(pulse_ctx, &pulse_phys_sources_cb, NULL);
+            o = pa_context_get_source_info_list(c, &pulse_phys_sources_cb, NULL);
             if (o)
                 pa_operation_unref(o);
+
+            pa_context_set_subscribe_callback(c, pulse_subscribecallback, userdata);
+            o = pa_context_subscribe(
+                c,
+                (pa_subscription_mask_t)(PA_SUBSCRIPTION_MASK_SINK | PA_SUBSCRIPTION_MASK_SOURCE),
+                NULL,
+                userdata);
+            if (o)
+                pa_operation_unref(o);
+
             break;
 
         case PA_CONTEXT_FAILED:
-- 
2.52.0

